yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -129,22 +147,60 @@ export class L1Publisher implements L2BlockReceiver {
    * @returns True once the tx has been confirmed and is successful, false on revert or interrupt, blocks otherwise.
    */
   public async processL2Block(block: L2Block): Promise<boolean> {
-    const txData = {
-      header: block.header.toBuffer(),
-      archive: block.archive.root.toBuffer(),
-      body: block.bodyToBuffer(),
-      proof: Buffer.alloc(0),
-    };
+    // TODO: Remove this block number check, it's here because we don't currently have proper genesis state on the contract
     const lastArchive = block.header.lastArchive.root.toBuffer();
+    if (block.number != 1 && !(await this.checkLastArchiveHash(lastArchive))) {
+      this.log(`Detected different last archive prior to publishing a block, aborting publish...`);
+      return false;
+    }
+
+    const encodedBody = block.bodyToBuffer();
 
+    // Publish block transaction effects
     while (!this.interrupted) {
-      // TODO: Remove this block number check, it's here because we don't currently have proper genesis state on the contract
-      if (block.number != 1 && !(await this.checkLastArchiveHash(lastArchive))) {
-        this.log(`Detected different last archive prior to publishing a block, aborting publish...`);
+      if (await this.txSender.checkIfTxsAreAvailable(block)) {
```

Comment:
This is nice 👍 Could be a neat test to check cost of sending an empty block of a size that have been sent before. The effects should be the same, so it is essentially just the header and proof you put in. Making it a viable option to have a small empty block in the backhand when sequencing 🤔 

Likely also something that @Cooper-Kunz would find interesting.